ui/src/main/webapp/components/MenuInput.tsx

```diff
@@ -108,37 +143,44 @@ function MenuInput({ handleRequestOpen }) {
-                        servicesGroup[group.groupName].push({
-                            name: serviceName,
+            groupsMenu.forEach(
+                (group: { groupServices: string[]; groupName: string; groupTitle: string }) => {
+                    if (group?.groupServices) {
+                        servicesGroup[group.groupName] = [];
+                        group.groupServices.forEach((serviceName: string) => {
+                            servicesGroup[group.groupName].push({
+                                name: serviceName,
+                                hasSubmenu: false,
+                                title:
+                                    services.find(
+                                        (service: { name: string }) => service.name === serviceName
+                                    )?.title || '', // what should be done when title empty
+                                subTitle: services.find(
+                                    (service: { name: string }) => service.name === serviceName
+                                )?.subTitle,
+                            });
+                        });
+                        servicesGroup[ROOT_GROUP_NAME].push({
+                            name: group.groupName,
+                            title: group.groupTitle,
+                            hasSubmenu: true,
+                        });
+                    } else {
+                        servicesGroup[ROOT_GROUP_NAME].push({
+                            name: group.groupName,
+                            title: group.groupTitle,
+                            subTitle: services.find(
+                                (service: { name: string }) => service.name === group.groupName
+                            )?.subTitle,
                             hasSubmenu: false,
-                            title: services.find((service) => service.name === serviceName).title,
-                            subTitle: services.find((service) => service.name === serviceName)
-                                ?.subTitle,
                         });
-                    });
-                    servicesGroup[ROOT_GROUP_NAME].push({
-                        name: group.groupName,
-                        title: group.groupTitle,
-                        hasSubmenu: true,
-                    });
-                } else {
-                    servicesGroup[ROOT_GROUP_NAME].push({
-                        name: group.groupName,
-                        title: group.groupTitle,
-                        subTitle: services.find((service) => service.name === group.groupName)
-                            ?.subTitle,
-                        hasSubmenu: false,
-                    });
+                    }
                 }
-            });
+            );
         } else {
-            servicesGroup[ROOT_GROUP_NAME] = services.map((service) => ({
+            servicesGroup[ROOT_GROUP_NAME] = services.map((service: Group) => ({
```

Comment:
```suggestion
            servicesGroup[ROOT_GROUP_NAME] = services.map((service) => ({
```
it is there
<img width="1069" alt="image" src="https://github.com/splunk/addonfactory-ucc-generator/assets/142901247/8e2415dd-0180-498c-9a12-7823cf2c1e3b">